src/main/java/com/google/cloud/tools/project/AppYaml.java

```diff
@@ -16,54 +16,113 @@
 
 package com.google.cloud.tools.project;
 
-import java.io.IOException;
 import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Collections;
 import java.util.Map;
 import javax.annotation.Nullable;
 import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
 
-/** Tools for reading app.yaml */
+/** Tools for reading {@code app.yaml}. */
 public class AppYaml {
 
-  private final Map<String, ?> yamlMap;
-
+  private static final String ENVIRONMENT_TYPE_KEY = "env";
   private static final String RUNTIME_KEY = "runtime";
+  private static final String API_VERSION_KEY = "api_version";
```

Comment:
Personally, I've never seen the point of replacing these types of strings with named constants. The field names are not more descriptive than the strings themselves